SPECS/xpmem-hwe/xpmem-hwe.spec

```diff
@@ -0,0 +1,312 @@
+%global release_suffix _6.12.40.1.1
+%else
+%global target_kernel_version_full f.a.k.e
+%endif
+
+%global KVERSION %{target_kernel_version_full}
+%global K_SRC /lib/modules/%{target_kernel_version_full}/build
+
+%{!?_mofed_full_version: %define _mofed_full_version 24.10-20%{release_suffix}%{?dist}}
+
+# %{!?KVERSION: %global KVERSION %(uname -r)}
+%{!?KVERSION: %global KVERSION %{target_kernel_version_full}}
+%global kernel_version %{KVERSION}
+%global krelver %(echo -n %{KVERSION} | sed -e 's/-/_/g')
+%{!?K_SRC: %global K_SRC /lib/modules/%{KVERSION}/build}
+# A separate variable _release is required because of the odd way the
+# script append_number_to_package_release.sh works:
+%global _release 1.2410068
+
+%bcond_with kernel_only
+
+%if %{with kernel_only}
+%undefine _debugsource_packages
+%global debug_package %{nil}
+%global make_kernel_only SUBDIRS=kernel
+%else
+%global make_kernel_only %{nil}
+%endif
+
+%define need_firmware_dir 0%{?euleros} > 0
+
+%if "%_vendor" == "openEuler"
+%global __find_requires %{nil}
+%endif
+
+Summary:	 Cross-partition memory
+Name:		 xpmem-hwe
+Version:	 2.7.4
+Release:	 20%{release_suffix}%{?dist}
+License:	 GPLv2 and LGPLv2.1
+Group:		 System Environment/Libraries
+Vendor:          Microsoft Corporation
+Distribution:    Azure Linux
+BuildRequires:	 automake autoconf
+URL:		 https://github.com/openucx/xpmem
+Source0:         https://linux.mellanox.com/public/repo/mlnx_ofed/24.10-0.7.0.0/SRPMS/xpmem-2.7.4.tar.gz#/xpmem-%{version}.tar.gz
+ExclusiveArch:   aarch64
+
+# name gets a different value in subpackages
+%global _name %{name}
+%global _kmp_rel %{release}%{?_kmp_build_num}%{?_dist}
+# Required for e.g. SLES12:
+%if %{undefined make_build}
+%global make_build %{__make} %{?_smp_mflags}
+%endif
+
+# Ugly workaround until anolis kmod package stops requiring
+# 'kernel(' dependencies its kernel package does not provide.
+# This uses the __find_provides from /usr/lib/rpm/redhat/macros
+# rather than the one from /usr/lib/rpm/macros.d/macros.kmp
+%if 0%{?anolis} > 0
+%{?filter_setup}
+%endif
+
+BuildRequires:  gcc
+BuildRequires:  make
+BuildRequires:  kernel-hwe-devel = %{target_kernel_version_full}
+BuildRequires:  binutils
+BuildRequires:  systemd
+BuildRequires:  kmod
+BuildRequires:  mlnx-ofa_kernel-hwe-devel = %{_mofed_full_version}
+BuildRequires:  mlnx-ofa_kernel-hwe-source = %{_mofed_full_version}
+
+Requires:       mlnx-ofa_kernel-hwe = %{_mofed_full_version}
+Requires:       mlnx-ofa_kernel-hwe-modules = %{_mofed_full_version}
+Requires:       kernel-hwe = %{target_kernel_version_full}
+Requires:       kmod
+
+
+%description
+XPMEM is a Linux kernel module that enables a process to map the
+memory of another process into its virtual address space. Source code
+can be obtained by cloning the Git repository, original Mercurial
+repository or by downloading a tarball from the link above.
+
+This package includes helper tools for the kernel module.
+
+%if ! %{with kernel_only}
+%package -n libxpmem
```

Comment:
@PawelWMS should I rename this to `libxpmem-hwe`?

Reply:
Yes, please. Good catch btw., I missed it. We shouldn't have any same-named packages coming from different spec file.

Reply:
It might be helpful to define a macro to re-use within the spec like `%global kernel_suffix -hwe` or something like that to make it harder to do a typo. And it'll make it easier in the future to adjust these specs, when the toolkit is more flexible to handle the different kernel we have.

Reply:
Done. Thanks!